community/php7/APKBUILD

```diff
@@ -157,203 +322,285 @@ build() {
+	pkgdesc="PHP$_suffix Common Gateway Interface"
 
-	mkdir -p "$subpkgdir"/usr/bin
-	mv "$pkgdir"/usr/bin/php-cgi7 "$subpkgdir"/usr/bin/
-	ln -s php-cgi7 "$subpkgdir"/usr/bin/php-cgi
+	_mv "$pkgdir"/usr/bin/php-cgi$_suffix "$subpkgdir"/usr/bin/
+
+	[ "$_default_php" = yes ] \
+		&& ln -s php-cgi$_suffix "$subpkgdir"/usr/bin/php-cgi
 }
 
 fpm() {
-	pkgdesc="PHP FastCGI Process Manager"
-	depends="$pkgname-common"
+	pkgdesc="PHP$_suffix FastCGI Process Manager"
 
-	mv "$pkgdir"/var "$subpkgdir"/ || return 1
+	cd "$pkgdir"
 
-	mkdir -p "$subpkgdir"/etc/$pkgname \
-		"$subpkgdir"/usr \
-		"$subpkgdir"/var/lib \
-		"$subpkgdir"/var/log/$pkgname
-
-	mv "$pkgdir"/usr/share/$pkgname/fpm "$subpkgdir"/var/lib/$pkgname/ || return 1
-	mv "$pkgdir"/usr/sbin "$subpkgdir"/usr/ || return 1
-	mv "$pkgdir"/etc/$pkgname/php-fpm* "$subpkgdir"/etc/$pkgname/ || return 1
+	_mv var "$subpkgdir"/
+	_mv usr/share/$pkgname/fpm "$subpkgdir"/var/lib/$pkgname/
+	_mv usr/sbin "$subpkgdir"/usr/
+	_mv etc/$pkgname/php-fpm* "$subpkgdir"/etc/$pkgname/
 
 	local file; for file in php-fpm.conf php-fpm.d/www.conf; do
 		mv "$subpkgdir"/etc/$pkgname/$file.default \
-			"$subpkgdir"/etc/$pkgname/$file || return 1
+			"$subpkgdir"/etc/$pkgname/$file
 	done
 
-	install -D -m755 "$srcdir"/$pkgname-fpm.initd "$subpkgdir"/etc/init.d/php-fpm7 || return 1
-	install -D -m644 "$srcdir"/$pkgname-fpm.logrotate "$subpkgdir"/etc/logrotate.d/php-fpm7
+	install -D -m 755 "$srcdir"/$pkgname-fpm.initd \
+		"$subpkgdir"/etc/init.d/php-fpm$_suffix
+
+	install -D -m 644 "$srcdir"/$pkgname-fpm.logrotate \
+		"$subpkgdir"/etc/logrotate.d/php-fpm$_suffix
+
+	mkdir -p "$subpkgdir"/var/log/$pkgname
 }
 
 pear() {
-	pkgdesc="PHP Extension and Application Repository"
+	pkgdesc="PHP$_suffix Extension and Application Repository"
 	depends="$pkgname $pkgname-xml"
 
-	mkdir -p "$subpkgdir"/usr/bin \
-		"$subpkgdir"/etc/$pkgname
+	cd "$pkgdir"
+
+	# pecl needs xml extension and since we build it as shared, it must be
+	# explicitly declared to be loaded.
+	sed -i 's/\$INCARG/& -d extension=xml.so/' usr/bin/pecl
 
-	local file; for file in pecl pear; do
-		mv "$pkgdir"/usr/bin/$file "$subpkgdir"/usr/bin/
+	mkdir -p "$subpkgdir"/usr/bin
+	local file; for file in pecl pear peardev; do
+		mv usr/bin/$file "$subpkgdir"/usr/bin/$file$_suffix
+		[ "$_default_php" = yes ] \
+			&& ln -s $file$_suffix "$subpkgdir"/usr/bin/$file
 	done
-	mv "$pkgdir"/etc/$pkgname/pear.conf "$subpkgdir"/etc/$pkgname/ || return 1
-	mv "$pkgdir"/usr/share "$subpkgdir"/usr/
+
+	_mv etc/$pkgname/pear.conf "$subpkgdir"/etc/$pkgname/
+	_mv usr/share "$subpkgdir"/usr/
 }
 
 common() {
 	pkgdesc="$pkgdesc (common config)"
 	depends=""
 
-	mkdir -p "$subpkgdir"/usr
-	mv "$pkgdir"/usr/lib "$subpkgdir"/usr/ || return 1
-	mv "$pkgdir"/etc "$subpkgdir"/ || return 1
+	cd "$pkgdir"
 
-	mkdir "$subpkgdir"/etc/$pkgname/conf.d
+	_mv usr/lib "$subpkgdir"/usr/
+	_mv etc "$subpkgdir"/
+	mkdir -p "$subpkgdir"/$_extension_confd
 }
 
 phar() {
```

Comment:
phar scripts used rarely then phar extension. So I assumed not to pull unneeded stuff with it. And the extension subpackage must be clean enough: only two files - binary module and config that enable it.